driver-core/src/main/com/mongodb/internal/dns/DefaultDnsResolver.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.internal.dns;
+
+import com.mongodb.MongoClientException;
+import com.mongodb.MongoConfigurationException;
+
+import javax.naming.Context;
+import javax.naming.NamingEnumeration;
+import javax.naming.NamingException;
+import javax.naming.directory.Attribute;
+import javax.naming.directory.Attributes;
+import javax.naming.directory.InitialDirContext;
+import java.util.ArrayList;
+import java.util.Hashtable;
+import java.util.List;
+
+import static java.lang.String.format;
+import static java.util.Arrays.asList;
+
+/**
+ * Utility class for resolving SRV and TXT records.
+ *
+ * <p>This class should not be considered a part of the public API.</p>
+ */
+public final class DefaultDnsResolver implements DnsResolver {
```

Comment:
Unfortunate this is showing up as a new file.  But it's the same as what used to be DnsResolver, except that the methods are now non-static and implement an interface.  So that it's mockable.

Reply:
Ack